### ISSUE TITLE
Feature suggestion: optionally styled dropdownlist items?

### ISSUE THREAD

**Issue by vincentsels** [ericgio/react-bootstrap-typeahead#4]
In our project, some of the dropdownlist data entries should be styled differently. Concretely: we're displaying 5 'favorites' or 'recently used' items on top.

A clean solution might be to optionally allow a React `style` object on each data entry; next to the `id` and `name`? For instance:

```jsx
var myData = [
  {id: 1, name: 'John', style: {fontWeight: 'bold'}},
  {id: 2, name: 'Miles', style: {fontWeight: 'bold'}},
  {id: 3, name: 'Charles'},
  {id: 4, name: 'Herbie'},
];
```

If you agree with this, I'll gladly create a pull request.

**Comment by vincentsels** [comment]
Or alternatively, allow specifying a class name:

```jsx
var myData = [
  {id: 1, name: 'John', className: 'bold'},
  {id: 2, name: 'Miles', className: 'bold'},
  {id: 3, name: 'Charles'},
  {id: 4, name: 'Herbie'},
];
```

Or both :).

**Comment by ericgio** [comment]
Good suggestion, I'll think on it some more. Seems reasonable to be able to pass props to the `MenuItem`. I also want to provide a way to render more customized menu items (eg: display a profile photo).

**Comment by ericgio** [comment]
I added a `renderMenuItem` prop that provides complete control for rendering of the individual menu items: https://github.com/ericgio/react-bootstrap-typeahead/blob/master/README.md#rendermenuitem

This hook provides a lot of flexibility, but may be more than you need, especially since you may lose some default behavior. I'm thinking of adding a more basic hook that allows you to control rendering of just the menu item contents. Let me know what you think.

**Comment by ericgio** [comment]
Actually `renderMenuItem` felt like overkill, so I changed it to `renderMenuItemChildren`, which allows you to customize the contents of the menu item while keeping most of the default behavior. This is now available in v0.3.0
